ibis/config.py

```diff
@@ -76,6 +79,7 @@ class Interactive(Config):
     max_string: int = 80
     max_depth: int = 1
     show_types: bool = True
+    show_count: bool = True
```

Comment:
I think this is a nice feature to have.  I do not think it should default to true.

Interactive mode is supposed to be interactive, and something that can be computationally expensive isn't something we want to turn on by default.

The main failure case I can think of here is reading from CSVs in a blob store.  With `show_count=False`, this can be done with a range request and only download the first N rows (ish), except if there's an order-by.  `show_count=True` would mean any operation would trigger a download of the entire file.

And yes, that is one more reason not to use CSVs, but :shrug:

Reply:
OK, I am OK with defaulting to False. Let's just wait for Philip to chime in to verify, but I assume he is going to have the same opinion as you.

I really don't think it is worth the complexity/cleverness, but throwing it out there that we could make each relation have its own config, and that could be auto-determined based on if it's a CSV/parquet/physical table, local/remote, etc.

Reply:
> I assume he is going to have the same opinion as you.

We are slowly merging into the same person.  The transformation is nearly complete.

> I really don't think it is worth the complexity/cleverness, but throwing it out there that we could make each relation have its own config, and that could be auto-determined based on if it's a CSV/parquet/physical table, local/remote, etc.

I mean, it's definitely a reliably cheap(er) operation on anything that isn't a CSV, but I agree this seems like too much complexity.  I think a better move is your suggestion of persistent default configuration

Reply:
Yes, let's leave the default to `False`, otherwise it's a huge performance footgun/bug magnet.